src/hooks/useTable.js

```diff
@@ -139,14 +139,16 @@ export const useTable = (props, ...plugins) => {
   // This function is absolutely necessary and MUST be called on
   // any rows the user wishes to be displayed.
   api.prepareRow = row => {
-    const { index } = row
+    const { path, index } = row
     row.getRowProps = props =>
       mergeProps(
-        { key: ['row', index].join('_') },
+        { key: ['row', path || index].join('_') },
         applyHooks(api.hooks.getRowProps, row, api),
         props
       )
 
+    // need to apply any row specific hooks (useExpanded requires this)
+    applyHooks(api.hooks.row, row, api)
```

Comment:
I could have sworn we were already executing a `row` specific hook, were we not?